datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/handler/DatalakeDatabaseRestoreWaitHandler.java

```diff
@@ -68,8 +80,20 @@ protected Selectable doAccept(HandlerEvent<DatalakeDatabaseRestoreWaitRequest> e
             LOGGER.info("Database restore poller stopped for cluster: {}", sdxId);
             sdxBackupRestoreService.updateDatabaseStatusEntry(event.getData().getOperationId(),
                     SdxOperationStatus.FAILED, pollerStoppedException.getLocalizedMessage());
+            // Get at which step the timeout happens.
+            SdxCluster sdxCluster = sdxClusterRepository.findById(sdxId).orElseThrow(notFound("SDX cluster", sdxId));
```

Comment:
Since this is identical to the logic chunk above, we should ideally move this to a central location and just call it from there. You can decide the best way to do this.

Reply:
Hi @sxxgrc , I refactored that into [SdxBackupRestoreService.java](https://github.com/hortonworks/cloudbreak/pull/14324/files#diff-ab042d3d14cadab0aa6559b87f5e57f79a0641839b41a933aa3e83643aec8810), please let me know if it looks good. Thank you!

Reply:
Looks good Grace thank you.